### PR TITLE
[WIP] provider/aws: AWS DeviceFarm

### PR DESCRIPTION
**Pull request by stack72**
Project
== 
- [x] schema
- [x] CRUD functionality
- [x] acceptance tests
- [x] documentation

```hcl
make testacc TEST=./builtin/providers/aws TESTARGS='-run=DeviceFarmProject' 2>~/tf.log
go generate ./...
TF_ACC=1 go test ./builtin/providers/aws -v -run=DeviceFarmProject -timeout 90m
=== RUN   TestAccAWSDeviceFarmProject_basic
--- PASS: TestAccAWSDeviceFarmProject_basic (7.61s)
=== RUN   TestAccAWSDeviceFarmProject_update
--- PASS: TestAccAWSDeviceFarmProject_update (12.08s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	19.709s
```

Run
== 
- [ ] schema
- [ ] CRUD functionality
- [ ] acceptance tests
- [ ] documentation

Device Pool
== 
- [x] schema
- [x] CRUD functionality
- [ ] acceptance tests
- [ ] documentation

### CODE REVIEW

File: builtin/providers/aws/resource_aws_devicefarm_project.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+func resourceAwsDeviceFarmProjectCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).devicefarmconn
+	region := meta.(*AWSClient).region
+
+	//	We need to ensure that DeviceFarm is only being run against us-west-2
+	//	As this is the only place that AWS currently supports it
+	if region != "us-west-2" {
+		return fmt.Errorf("DeviceFarm can only be used with us-west-2. You are trying to use it on %s", region)
```
Comment (apparentlymart) [inline builtin/providers/aws/resource_aws_devicefarm_project.go]: If we were to skip this test and the user set it to some other region, would the AWS API fail or would it do something unwanted?

If it'll fail anyway, maybe it's better to just skip this check and let the API call fail below, and then Terraform won't need to be updated if this service gets extended to work in other regions.

Comment (stack72) [inline builtin/providers/aws/resource_aws_devicefarm_project.go]: The provider just hangs until timeout. It took me quite a while before it showed me this:

```
make testacc TEST=./builtin/providers/aws TESTARGS='-run=DeviceFarm' 2>~/tf.log
go generate ./...
TF_ACC=1 go test ./builtin/providers/aws -v -run=DeviceFarm -timeout 90m
=== RUN   TestAccAWSDeviceFarmProject_basic
--- FAIL: TestAccAWSDeviceFarmProject_basic (448.13s)
	testing.go:137: Step 0 error: Error applying: 1 error(s) occurred:

		* aws_devicefarm_project.foo: Error creating DeviceFarm Project: RequestError: send request failed
		caused by: Post https://devicefarm.eu-west-1.amazonaws.com/: dial tcp 81.200.64.50:443: i/o timeout
FAIL
exit status 1
FAIL	github.com/hashicorp/terraform/builtin/providers/aws	448.149s
```

**Comment by catsby** [comment]
Hey @stack72 is this still WIP? Do you need someone to take over or should we close for now?

**Comment by stack72** [comment]
@catsby thanks for the reminder on this - will close this for now as it's a WIP but will continue adding to it

**Pull request closed without merging**